events/source/equeue.c

```diff
@@ -21,6 +21,12 @@
 #include <stdint.h>
 #include <string.h>
 
+// check if the event is allocaded by user - event address is outside queues internal buffer address range
+#define IS_USER_ALLOCATED_EVENT(e) (((unsigned int)(e) < (unsigned int)q->buffer) || ((unsigned int)(e) > ((unsigned int)q->slab.data)))
```

Comment:
Casting to integer makes it probably safer than comparing pointers (it's undefined behaviour to compare pointers not inside the same object), but the integers should be `uintptr_t`, or this could go nasty on 64-bit platforms.

Reply:
fixed `unsigned int` -> `uintptr_t`